src/CashuWallet.ts

```
@@ -364,22 +429,31 @@ class CashuWallet {
 
 	/**
 	 * Creates a split payload
-	 * @param amount1 amount to keep
-	 * @param amount2 amount to send
-	 * @param proofsToSend proofs to split
+	 * @param amount amount to send
+	 * @param proofsToSend proofs to split*
+	 * @param preference optional preference for splitting proofs into specific amounts. overrides amount param
+	 * @param count? optionally set count to derive secret deterministically. CashuWallet class must be initialized with seed phrase to take effect
 	 * @returns
 	 */
 	private createSplitPayload(
 		amount: number,
 		proofsToSend: Array<Proof>,
-		preference?: Array<AmountPreference>
+		preference?: Array<AmountPreference>,
+		count?: number
 	): {
 		payload: SplitPayload;
 		blindedMessages: BlindedTransaction;
 	} {
 		const totalAmount = proofsToSend.reduce((total, curr) => total + curr.amount, 0);
-		const keepBlindedMessages = this.createRandomBlindedMessages(totalAmount - amount);
-		const sendBlindedMessages = this.createRandomBlindedMessages(amount, preference);
+		const keepBlindedMessages = this.createRandomBlindedMessages(
+			totalAmount - amount,
+			undefined,
+			count
+		);
+		if (this._seed && count) {
+			count = count + keepBlindedMessages.secrets.length;
+		}
```

We should probably throw an error if `count` is provided, but `_seed` is not.